engines/osxnative/resultset.go

```diff
@@ -97,3 +96,18 @@ func (r resultset) ExtractFolder(path string, handler engines.FileHandler) error
 		return nil
 	})
 }
+
+func (r resultset) Dispose() error {
+	err := r.taskUser.delete()
+	if err != nil {
```

Comment:
I am skipping user creation tests because they are quite difficult to implement. Going to address this in a follow up patch.